src/zope/testrunner/interfaces.py

```diff
@@ -100,3 +100,60 @@ class ITestRunner(zope.interface.Interface):
 
     options = zope.interface.Attribute(
       "Provides access to configuration options.")
+
+
+class IMinimalTestLayer(zope.interface.Interface):
+    """A test layer.
+
+    This is the bare minimum that a ``layer`` attribute on a test suite
+    should provide.
+    """
+
+    __bases__ = zope.interface.Attribute(
+        "A tuple of base layers.")
+
+    __name__ = zope.interface.Attribute(
+        "Name of the layer")
+
+    __module__ = zope.interface.Attribute(
+        "Dotted name of the module that defines this layer")
+
+    def __hash__():
+        """A test layer must be hashable"""
```

Comment:
I tried to add \__eq__ and it felt wrong to me again.

I don't oppose documenting that, but I can't find the words myself.